### PR TITLE
Handle rate exceeded

### PR DESCRIPTION
**Pull request by Hehk**
- TS conversion so I could have lambda's types come through to guide some exploration
- Add retry logic to `invoke` this way it silently paints over these errors that are mostly outside of our control

### CODE REVIEW

**Comment by alif** [comment]
Nitpick: The whitespacing in the changes and generally in this repo are kind of messy. Introducing `prettier` would be good to get consistent with other codebases.

File: lib/util.ts [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+}
+
+export function ensureDir(dir:string) : void {
+  const parent = path.dirname(dir)
+  fs.existsSync(parent) || ensureDir(parent)
+  fs.existsSync(dir) || fs.mkdirSync(dir)
+}
+
+function isRetryableError (error : Error) {
+  if (error.message.includes('Rate Exceeded.')) {
```
Comment (alif) [inline lib/util.ts:117]: `return error.message.includes('Rate Exceeded.')`

File: lib/util.ts [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+export function readFile (p : string, encoding : BufferEncoding = 'utf8') : string {
+  ensureDir(path.dirname(p))
+  if (!fs.existsSync(p)) return ''
+  return fs.readFileSync(p, encoding)
+}
+
+export async function readFileAsync(p : string, encoding : BufferEncoding = 'utf8') : Promise<string | undefined> {
+  ensureDir(path.dirname(p))
+  return new Promise((res) => {
+    fs.readFile(p, encoding, (_err, data) => res(data))
```
Comment (alif) [inline lib/util.ts:99]: I'm confused why we are not using `readFileSync` in this and rename the function

Comment (Hehk) [reply lib/util.ts:99]: I am not really sure either. This is an old util, I did not want to change it incase if something was depending on some edge

**Pull request merged**